tests/ftests/088-sudo-cgclassify_systemd_scope.py

```diff
@@ -0,0 +1,169 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: LGPL-2.1-only
+#
+# Test to replace idle_thread of scope unit using cgclassify -r
+#
+# Copyright (c) 2023 Oracle and/or its affiliates.
+# Author: Kamalesh Babulal <kamalesh.babulal@oracle.com>
+#
+
+from process import Process
+from cgroup import Cgroup
+from run import RunError
+import consts
+import ftests
+import time
+import sys
+import os
+
+CONTROLLER = 'cpu'
+SLICE = 'libcgroup.slice'
+CGNAME = os.path.join(SLICE, '086cgcreate.scope')
+OUT_OF_SCOPE_CGNAME1 = '086outofscope'
+OUT_OF_SCOPE_CGNAME2 = '086outofscope.scope'
+
+
+def prereqs(config):
+    result = consts.TEST_PASSED
+    cause = None
+
+    if config.args.container:
+        result = consts.TEST_SKIPPED
+        cause = 'This test cannot be run within a container'
+
+    return result, cause
+
+
+def setup(config):
+    pass
+
+
+def test(config):
+    result = consts.TEST_PASSED
+    cause = None
+
+    #
+    # Test 1: call cgclassify -r to replace idle_thread with infinite while loop
+    #
+    Cgroup.create_and_validate(config, CONTROLLER, CGNAME, create_scope=True)
+    idle_pid = Cgroup.get_pids_in_cgroup(config, CGNAME, CONTROLLER)[0]
+
+    config.process.create_process_in_cgroup(config, CONTROLLER, CGNAME, replace_idle=True)
+
+    # We need pause, before the cgroups.procs gets updated, post cgclassify
+    time.sleep(1)
+
+    pid = Cgroup.get_pids_in_cgroup(config, CGNAME, CONTROLLER)
+    if len(pid) > 1 and idle_pid == pid[0]:
+        result = consts.TEST_FAILED
+        cause = 'Failed to replace scope idle_thread pid {}'.format(idle_pid)
+        return result, cause
+
+    #
+    # Test 2: call cgclassify -r to replace non-idle_thread (while loop,
+    #         previously created) with another while loop.
+    #
+    try:
+        config.process.create_process_in_cgroup(config, CONTROLLER, CGNAME, replace_idle=True)
+    except RunError as re:
+        if "Failed to find idle_thread task" not in re.stderr:
+            raise re
```

Comment:
We probably need an `else` here to handle the case where the create_process erroneously succeeds

Reply:
Make sense will add the `else` block too for erroneous success.